### PR TITLE
Add a migration guide for the YAML pipeline editor

### PR DESCRIPTION
**Pull request by harrietgrace**
A sketchy outline of a migration guide for the UI to YAML editor change. I'll keep this as a draft until it's ready for review 🌈

### CODE REVIEW

**Comment by plaindocs** [comment]
Love that last commit message!

**Comment by harrietgrace** [comment]
Needs a sidebar link, but otherwise, this one is ready for a review @plaindocs 😊

File: pages/tutorials/pipeline_upgrade.md.erb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+```
+{: codeblock-file="pipeline.yml"}
+
+Prior to migrating to YAML Steps, the command will echo `pipeline`.
+
+## Using YAML Steps for new pipelines
+
+To use the YAML Steps editor for new pipelines created in your organization, you'll need to opt-in on the Pipeline YAML Migration page in Organization Settings. 
+
+<%= image "bulk-migration.png", width: 1744/2, height: 446/2, alt: "Screenshot of the button to use YAML steps for new pipelines" %>
```
[the remark this answers is not in the document]
Comment (harrietgrace) [reply pages/tutorials/pipeline_upgrade.md.erb:48]: It is the wrong image, I don't have the screenshot of that one yet so I stuck that there to place-hold 😄

File: pages/tutorials/pipeline_upgrade.md.erb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  LEVEL: "pipeline"
+
+steps:
+  command: "echo $LEVEL"
+  env:
+    LEVEL: "step"
+```
+{: codeblock-file="pipeline.yml"}
+
+Prior to migrating to YAML Steps, the command will echo `pipeline`.
```
[the remark this answers is not in the document]
Comment (harrietgrace) [reply pages/tutorials/pipeline_upgrade.md.erb:43]: What do you mean by detect?

File: pages/tutorials/pipeline_upgrade.md.erb [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Migrating to YAML Steps
+
+This guide will explain the differences between the web-based editor and the YAML Steps editor, and provides steps for completing the migration for your organization's pipelines. 
+
+{:toc}
+
+## What changes when you migrate
+
+Instead of creating and managing pipeline steps with the GUI-like step editor, pipelines will be managed in YAML in the YAML Steps editor.  
+
```
[the remark this answers is not in the document]
Comment (harrietgrace) [reply pages/tutorials/pipeline_upgrade.md.erb:10]: Yep totally! I've asked Fred to put together a combo image of the old one and the new one, and we can use it for the announcement too.

File: pages/tutorials/pipeline_upgrade.md.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (plaindocs) [reply pages/tutorials/pipeline_upgrade.md.erb:43]: I'm wondering if this is a thing that could be solved in Product Land with a feature request. :-p

File: pages/tutorials/pipeline_upgrade.md.erb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+<%= image "individual-migration.png", width: 1730/2, height: 1086/2, alt: "Screenshot of the YAML Steps conversion panel and button" %>
+
+Click this button to convert your pipeline steps into YAML. The new YAML steps version of your pipeline will be auto-saved during migration, and will replace the web-based steps.
+
+If you expect to have compatibility issues with your environment variables, individual migration is recommended. 
+
+### Bulk migration
+
+You can migrate all of the pipelines in your organization at the same time using the bulk migration button on the migration page.
+
```
Comment (plaindocs) [inline pages/tutorials/pipeline_upgrade.md.erb:70]: <section class="Docs__note">
<p>If you expect to have < a href="#compatibility-issues">compatibility issues</a> with your environment variables, migrate your pipelines < a href="#individually">individually</a>. 
</p>
</section>

File: pages/tutorials/pipeline_upgrade.md.erb [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+To use the YAML Steps editor for new pipelines created in your organization, you'll need to opt-in on the Pipeline YAML Migration page in Organization Settings. 
+
+<%= image "bulk-migration.png", width: 1744/2, height: 446/2, alt: "Screenshot of the button to use YAML steps for new pipelines" %>
+
+Clicking the button to start using YAML pipelines won't change anything for any existing pipelines in your organization. Migration of existing pipelines will need to be completed either individually or using the bulk migration tool. See [Steps to migrate existing pipelines](#steps-to-migrate-existing-pipelines) below for further information.  
+
+## Steps to migrate existing pipelines
+
+The migration page can be accessed by organization administrators from the Organization Settings page. On this page you can see which pipelines have already been migrated and which are yet to migrate.
```
[the remark this answers is not in the document]
Comment (harrietgrace) [reply pages/tutorials/pipeline_upgrade.md.erb:57]: Hrmm, I'm not actually sure! @eleanorakh do you need to be an admin to do the individual migration?

**Comment by harrietgrace** [comment]
@scopegate love the image! Is it retina? It looks a little on the small side?

File: pages/tutorials/pipeline_upgrade.md.erb [hunk 6]
```diff
@@ -0,0 +1,75 @@
+# Migrating to YAML Steps
+
+This guide will explain the differences between the web-based editor and the YAML Steps editor, and provides steps for completing the migration for your organization's pipelines. 
+
+{:toc}
+
+## What changes when you migrate
```
[the remark this answers is not in the document]
Comment (harrietgrace) [reply pages/tutorials/pipeline_upgrade.md.erb:7]: Yeah nice. I also moved the bit about migration that didn't really fit in that section out before that heading.

**Comment by scopegate** [comment]
It looks like there are still some specs failing here, related to some missing anchors on the Pipeline upgrade page. 

There are three anchors on the page that have links pointing to them, but aren't present on the page. I'd jump in and fix, but I'm not sure what they should be pointing to:

- `#steps-to-migrate-existing-pipelines`
- `#individually`
- `#compatibility-issues`

File: pages/tutorials/pipeline_upgrade.md.erb [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+Click this button to convert your pipeline steps into YAML.
+
+If you expect to have compatibility issues with your environment variables, individual migration is recommended. 
+
+### Bulk migration
+
+You can migrate all of the pipelines in your organization at the same time using the "Replace web steps for all pipelines with YAML" button on the migration page. The new YAML steps version of your pipeline will be auto-saved during migration, and will replace the web-based steps.
+
+<section class="Docs__note">
+<p>If you expect to have <a href="#compatibility-issues">compatibility issues</a> with your environment variables, migrate your pipelines <a href="#individually">individually</a>.
```
Comment (plaindocs) [inline pages/tutorials/pipeline_upgrade.md.erb:76]:
```suggestion
<p>If you expect to have <a href="#what-is-the-yaml-steps-editor-compatibility-issues">compatibility issues</a> with your environment variables, migrate your pipelines <a href="#migrating-existing-pipelines-individual-migration">individually</a>. 
```

File: pages/tutorials/pipeline_upgrade.md.erb [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+Prior to migrating to YAML Steps, the command will echo `pipeline`.
+
+## Using YAML Steps for new pipelines
+
+To use the YAML Steps editor for new pipelines created in your organization, you'll need to opt-in on the Pipeline YAML Migration page in Organization Settings. 
+
+<%= image "new-pipelines.png", width: 1754/2, height: 442/2, alt: "Screenshot of the button to use YAML steps for new pipelines" %>
+
+Clicking the button to start using YAML pipelines won't change anything for any existing pipelines in your organization. Migration of existing pipelines will need to be completed either individually or using the bulk migration tool. See [Steps to migrate existing pipelines](#steps-to-migrate-existing-pipelines) below for further information.
```
Comment (plaindocs) [inline pages/tutorials/pipeline_upgrade.md.erb:53]:
```suggestion
Clicking the button to start using YAML pipelines won't change anything for any existing pipelines in your organization. Migration of existing pipelines will need to be completed either individually or using the bulk migration tool. See [Steps to migrate existing pipelines](#migrating-existing-pipelines) below for further information.  
```

**Pull request merged**